src/RL/representation/default/heterogeneoustrajectorystate.jl

```diff
@@ -0,0 +1,109 @@
+using FillArrays
+using StatsBase:countmap
+
+"""
+    HeterogeneousTrajectoryState
+
+The most basic state representation, with a featured graph, the index of the variable to branch on and optionnaly a list of possible values.
+"""
+struct HeterogeneousTrajectoryState <: GraphTrajectoryState
+    fg::HeterogeneousFeaturedGraph
+    variableIdx::Int
+    possibleValuesIdx::Union{Nothing, Vector{Int64}}
+
+function HeterogeneousTrajectoryState(fg, var, val)
+    @assert var <= size(fg.varnf,2) "The variable index is out of bound"
+    @assert all(val .<= size(fg.valnf,2)) "One of the values index is out of bound"
+    @assert all([tuple[2] for tuple in countmap([c for c in val])] .== 1) "values array contains one element at least twice"
+    return new(fg, var, val)
+end
+end
+
+HeterogeneousTrajectoryState(sr::AbstractStateRepresentation) = throw(ErrorException("missing function HeterogeneousTrajectoryState(::$(typeof(sr)))."))
+
+"""
+    BatchedHeterogeneousTrajectoryState
+
+The batched version of the `HeterogeneousTrajectoryState`.
+
+It contains all the information that would be stored in a `HeterogeneousFeaturedGraph` but reorganised to enable simultaneous 
+computation on a few graphs.
+"""
+struct BatchedHeterogeneousTrajectoryState{T} <: NonTabularTrajectoryState
+    fg::BatchedHeterogeneousFeaturedGraph{T}
+    variableIdx::AbstractVector{Int}
+    possibleValuesIdx::Union{Nothing, AbstractVector{Vector{Int64}}} #Todo change this to abstract matrix
+
+function BatchedHeterogeneousTrajectoryState(fg, var, val)
+    @assert all(var .<= size(fg.varnf,2)) "The variable index is out of bound"
+    @assert all(reduce(vcat,val) .<= size(fg.valnf,2)) "One of the values index is out of bound"
+    @assert all(reduce(vcat,[[tuple[2] for tuple in countmap([c for c in vals])] for vals in val]) .== 1) "values array contains one element at least twice"
```

Comment:
Well, This is absolutely inefficient and must be removed for production, but I'd prefer having these kinds of verifications to prevent any anti-pattern behavior while trying to make our new pipeline work.